docs/pvc-recovery.md

```diff
@@ -0,0 +1,263 @@
+  - `oc set volume deployment/<recovery_node_name> --add -t persistentVolumeClaim --claim-name=<name of the old pvc> --name=elasticsearch-storage --overwrite`
+    - while not necessary, it is a good idea to match the PVC numbering to the node numbering for ease of remembering, e.g. `oc set volume deployment/elasticsearch-cdm-ap5obh1j-1 --add -t persistentVolumeClaim --claim-name=elasticsearch-elasticsearch-cdm-xm9nl5cb-1 --name=elasticsearch-storage --overwrite`
+
+1. Scale up all your recovery nodes
+  - `oc scale deployment --replicas=1 -l cluster-name=elasticsearch-recovery`
+  - `oc scale statefulset --replicas=<value before scaledown> -l cluster-name=elasticsearch-recovery`
+
+1. Wait for the recovery cluster to come up and ensure the prior indices exist.
+  - `oc exec -c elasticsearch <es_pod_name> -- indices`
+  - if you do not see your data, verify that you have correctly updated the `CLUSTER_NAME` env var name for each of your nodes.
+
+1. Update the recovery cluster to allow the ES service account to read its indices.
+  - Run this once on the recovery cluster (not for each node)
+  - `oc exec -c elasticsearch <recovery_pod_name> -- sed -i sgconfig/roles_mapping.yml -e "/system.admin/a \    - 'CN=logging-es,OU=OpenShift,O=Logging'"`
+  - `oc exec -c elasticsearch <recovery_pod_name> -- es_seed_acl`
+
+1. Update the Elasticsearch configmap to know how to reindex from remote
+  - `oc edit configmap elasticsearch`
+  - insert the following in `elasticsearch.yml` be sure to pay attention to indentation, `reindex` should be at the same indentation amount as `path`
+```
+    reindex:
+      remote.whitelist: "elasticsearch-recovery.openshift-logging.svc:9200"
+      ssl:
+        verification_mode: certificate
+        truststore:
+          path: /etc/elasticsearch/secret/truststore
+          password: tspass
+          type: jks
+        keystore:
+          path: /etc/elasticsearch/secret/key
+          type: jks
+          password: kspass
+```
+    - e.g.
+```
+    gateway:
+      recover_after_nodes: 2
+      expected_nodes: 3
+      recover_after_time: ${RECOVER_AFTER_TIME}
+
+    path:
+      data: /elasticsearch/persistent/${CLUSTER_NAME}/data
+      logs: /elasticsearch/persistent/${CLUSTER_NAME}/logs
+
+    reindex:
+      remote.whitelist: "elasticsearch-recovery.openshift-logging.svc:9200"
+      ssl:
+        verification_mode: certificate
+        truststore:
+          path: /etc/elasticsearch/secret/truststore
+          password: tspass
+          type: jks
+        keystore:
+          path: /etc/elasticsearch/secret/key
+          type: jks
+          password: kspass
+```
+
+1. Restart the cluster to pick up the configmap changes:
+  - `oc delete pod -l cluster-name=elasticsearch`
+
+1. Wait for the cluster to come back up
+
+1. Run the recovery script (in the appendix) from your local machine where you are running your other `oc` commands.
```

Comment:
An example `oc exec` would be nice! In addition a reference to the recovery cluster.

Reply:
this cannot be run from `oc exec` but ack on providing an example of running it